test/integration/addons_test.go

```diff
@@ -712,3 +713,84 @@ func validateGCPAuthAddon(ctx context.Context, t *testing.T, profile string) {
 		t.Errorf("failed disabling gcp-auth addon. arg %q.s %v", rr.Command(), err)
 	}
 }
+
+func validateAutoPause(ctx context.Context, t *testing.T, profile string) {
+	defer PostMortemLogs(t, profile)
+
+	client, err := kapi.Client(profile)
+	if err != nil {
+		t.Fatalf("failed to get Kubernetes client for %s: %v", profile, err)
+	}
+
+	// wait for auto-pause deployment
+	start := time.Now()
+	if err := kapi.WaitForDeploymentToStabilize(client, "auto-pause", "auto-pause-proxy", Minutes(6)); err != nil {
+		t.Errorf("failed waiting for auto-pause deployment to stabilize: %v", err)
+	}
+	t.Logf("auto-pause stabilized in %s", time.Since(start))
+
+	if _, err := PodWait(ctx, t, profile, "auto-pause", "app=auto-pause-proxy", Minutes(6)); err != nil {
+		t.Fatalf("failed waiting for app=auto-pause pod: %v", err)
+	}
+	// check for kubernetes status to be running
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "status", "--format={{.APIServer}}"))
+	if err != nil {
+		t.Fatalf("failed to get minikube status: %v", err)
+	}
+	want := "Running"
+	if !strings.Contains(rr.Stdout.String(), want) {
+		t.Errorf("expected api-server status be %q, but got *%s*", want, rr.Stdout.String())
+	}
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "status", "--format={{.Kubelet}}"))
+	if err != nil {
+		t.Fatalf("failed to get minikube status: %v", err)
+	}
+	if !strings.Contains(rr.Stdout.String(), want) {
+		t.Errorf("expected kubelet status be %q, but got *%s*", want, rr.Stdout.String())
+	}
+	// wait to get kubernetes to get paused
+	time.Sleep(70 * time.Second)
+	// check for kubernetes to be stopped
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "status", "--format={{.APIServer}}"))
+	if err != nil {
+		t.Fatalf("failed to get minikube status: %v", err)
+	}
+	want = "Paused"
+	if !strings.Contains(rr.Stdout.String(), want) {
+		t.Errorf("expected api-server status be %q, but got *%s*", want, rr.Stdout.String())
+	}
+	want = "Stopped"
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "status", "--format={{.Kubelet}}"))
+	if err != nil {
+		t.Fatalf("failed to get minikube status: %v", err)
+	}
+	if !strings.Contains(rr.Stdout.String(), want) {
+		t.Errorf("expected kubelet status be %q, but got *%s*", want, rr.Stdout.String())
+	}
+	// do some operation in kubernetes
+	_, err = Run(t, exec.CommandContext(ctx, "kubectl", "--context", profile, "get", "ns"))
+	if err != nil {
+		t.Fatalf("get namespace: %v", err)
+	}
+	// check for kubernetes to be running
+	rr, err = Run(t, exec.CommandContext(ctx, Target(), "-p", profile, "status", "--format={{.APIServer}}"))
```

Comment:
Would be nice if we added a helper func like `apiServerStatus` instead of having to duplicate this line many times throughout the test.

Same with `Kubelet` as well

Reply:
sure, will make the change

Reply:
Looks much more cleaner now, thanks!